### PR TITLE
Fast api conversion course and event

### PR DESCRIPTION
**Pull request by YuetingLiao**
**Issue**

Convert to FastAPI for user/course. 
Modified files: App.py, api_models.py, test_user_course.py
user/event feature has not been implemented.

**Test Modifications**

Run the test command line from yacs.n root directory:
src/api/tests/test.sh

**Test Procedure**

1. Pull from FastAPIConversion-course-and-event branch
2. Run the command line above

### CODE REVIEW

File: src/api/app.py [hunk 1]
```diff
@@ -205,11 +205,19 @@ async def uploadHandler(
 #
 #     return user_controller.get_user_info(session_id)
 #
-#
-# @app.route('/api/user', methods=['POST'])
-# def add_user():
-#     return user_controller.add_user(request.json)
-#
+
```
Comment (lhain08) [inline src/api/app.py:208]: For the sake of letting the owner of this endpoint complete it themselves, undo these changes

**Review by lhain08 — CHANGES_REQUESTED** [review]

File: src/api/app.py (hunk 1 above)
Comment (YuetingLiao) [reply src/api/app.py:208]: I don't see it locally, but I pulled from FastAPIConversion branch again and merged.
Let me know if it's good to resolve.

**Review by lhain08 — APPROVED** [review]
